src/Http/Session.php

```diff
@@ -688,4 +683,39 @@ protected function _timedOut(): bool
 
         return $result;
     }
+
+    /**
+     * Allow to set session timeout before session is started in Middleware.
+     * Setting it in bootstrap when fetching the value from database result in query not showing DebugKit
+     * and other issues due to using ORM before bootstrapping of other plugins.
+     *
+     * @param int $lifetime in seconds
+     * @return void
+     * @throws \Cake\Core\Exception\CakeException
+     */
+    public function setSessionLifetime(int $lifetime): void
```

Comment:
The issue would be that this config variable is only read in constructor, and also have some side-effects when setting it.

So using setConfig would not set the `session.gc_maxlifetime`, and adding it would also require to change the `_timedOut()` to read directly from config.

Reply:
Yes, all the code in the constructor (barring last 2 lines probably) would be moved into the new `setConfig()` method and then the constructor can call that method.

Reply:
Thinking about this again, having a separate method to update session time time is good enough since we already have separate public methods for ini config `options()` and the handler `engine()`.